R/vulnerability/disasters-emergencies/flooding/2022-interim/england/service-availability/flooding-exposure-emergency-services.R

```diff
@@ -12,92 +17,194 @@ flood_risk_oa <- read_rds("data/vulnerability/disasters-emergencies/flooding/flo
+    osm_id %in% fire_reduce$osm_id & osm_id %in% ambo_reduce$osm_id ~ "ambo_fire",
+    osm_id %in% police_reduce$osm_id & osm_id %in% ambo_reduce$osm_id ~ "ambo_police",
+    osm_id %in% police_reduce$osm_id & osm_id %in% fire_reduce$osm_id ~ "fire_police",
+    osm_id %in% fire_reduce$osm_id ~ "fire",
+    osm_id %in% ambo_reduce$osm_id ~ "ambo",
+    osm_id %in% police_reduce$osm_id ~ "police",
+    TRUE ~ NA_character_
+  ))
+
+# Check service counts
+services |>
+  st_drop_geometry() |>
+  group_by(service) |>
+  summarise(count = n())
+# ambo: 575, ambo_fire: 12, fire: 1920, police: 2001
+
+services |>
+  st_drop_geometry() |>
+  group_by(osm_id) |>
+  mutate(count_id = n()) |>
+  filter(count_id > 1)
+
+# Check count of the geometry types
+services |>
+  mutate(area = st_area(geometry)) |>
+  st_drop_geometry() |>
+  mutate(point_poly = if_else(area == units::as_units(0, value = "m^2"), "point", "poly")) |>
+  group_by(point_poly) |>
+  summarise(count = n())
+
+# Keep only those within England boundary ----
+# Not this will cause duplicated polygon/multipolygon rows where it crosses multiple OAs
+services_eng <- services |>
+  st_transform(crs = st_crs(boundaries_oa_eng)) |>
+  st_join(boundaries_oa_eng, left = FALSE)
+
+services_eng |>
+  st_drop_geometry() |>
+  group_by(service) |>
+  summarise(count = n_distinct(osm_id))
+# ambo: 502, ambo_fire: 1, fire: 1520, police: 1462
+
+# Checking those with no OA match ----------------
+# Are in Ireland, Scot, Wales & France (picked up do to boundary box in line 21)
+services |>
+  st_transform(crs = st_crs(boundaries_oa_eng)) |>
+  st_join(boundaries_oa_eng) |>
+  mutate(no_oa_match = if_else(is.na(OA11CD), T, F)) |>
+  ggplot() +
+  geom_sf(aes(colour = no_oa_match))
+
+# Checking reasonableness of total numbers of each service -----
+# Source: https://www.gov.uk/government/statistical-data-sets/fire-statistics-data-tables
+# Fire: 'FIRE1403: Fire stations and appliances, by fire and rescue authority tab'
+# In 2020 1,393 Fire Stations in England
+
+# https://www.ukcrimestats.com/Police_Stations/
+# 355 Police stations
+
+# https://www.londonambulance.nhs.uk/talking-with-us/how-to-find-us/
+# 70 Ambulance stations in London
+
+# Check when multiple buildings of same type per OA 
+# Assumption: unlikely will be building for same service with same OA so assume is duplicate
+services_eng_dups <- services_eng |>
+  group_by(OA11CD, service) |>
+  mutate(count_id = n()) |>
+  filter(count_id > 1) |>
+  arrange(desc(count_id), OA11CD, name) |>
+  st_transform(crs = 4326)
+
+# Take top one where has name (if not null for all) and then the largest size
+services_eng_dedup <- services_eng_dups |>
+  mutate(size = st_area(geometry)) |>
+  group_by(OA11CD, service) |>
+  arrange(OA11CD, name, desc(size)) |>
+  slice(1)
+
+services_eng_dedup <- services_eng |>
+  filter(!osm_id %in% services_eng_dups$osm_id) |>
+  bind_rows(services_eng_dedup)
```

Comment:
Yeah suppose difficult to validate the assumption and there may be exceptions where is more than 1 of a service in a single OA but thinking was since there are about 171k OAs in England and if take fire (which has more than police or ambulance) there are 1,400 stations of these so if well distributed in theory unlikely to be crossover (and OAs are about ~125 households and have a population of ~300). Suppose balance of this assumption, which may have some exceptions, with the duplicated cases within the OSM data which would need manually checking perhaps.

Reply:
Have put this deduplication part into a separate function `osm_oa_deduping()` in case want to use logic in`osm_data_reduce()` and `osm_oa_deduping()` separately. Has been updated in https://github.com/britishredcrosssociety/resilience-index/pull/13/commits/558a30a28c83090e13a6f84dbf6db4b2d3de49ce.